app/org/sagebionetworks/bridge/models/schedules/ActivityScheduler.java

```diff
@@ -42,21 +41,36 @@ protected DateTime getScheduledTimeBasedOnEvent(ScheduleContext context) {
     protected void addScheduledActivityForAllTimes(List<ScheduledActivity> scheduledActivities, 
             SchedulePlan plan, ScheduleContext context, DateTime scheduledTime) {
         if (schedule.getTimes().isEmpty()) {
-            // We're using whatever hour/minute/seconds were in the original event.
+            scheduledTime = adjustOnceIntervalActivityWithNoTimes(context, scheduledTime);
             addScheduledActivityAtTime(scheduledActivities, plan, context, scheduledTime);
         } else {
             for (LocalTime time : schedule.getTimes()) {
-                scheduledTime = new DateTime(scheduledTime, context.getZone()).withTime(time);
+                scheduledTime = scheduledTime.withTime(time);
                 addScheduledActivityAtTime(scheduledActivities, plan, context, scheduledTime);
             }
         }
     }
     
+    /**
+     * BRIDGE-1589: Adjust one-time interval tasks with no specified times so the time portion is 
+     * midnight of the event day. We are curently using the event's time, but that time changes 
+     * with daylight savings time (unlike schedules with specified times... these times are set the 
+     * same regardless of time zone). It is accompanied by a validation rule to prevent setting 
+     * expiration of a one-time to less than 24 hours because that isn't going to make sense in 
+     * the rare event someone tries it.
+     */
+    protected DateTime adjustOnceIntervalActivityWithNoTimes(ScheduleContext context, DateTime scheduledTime) {
+        // We already know there are no times... but in case this is ever called in another sequence, do check it.
+        // Normalize to UTC minus one day from enrollment, to ensure no matter where user is at the time of request,
+        // the one-time activity is available.
+        if (Schedule.isScheduleWithoutTimes(schedule)) {
+            return Schedule.eventToMidnight(scheduledTime);
+        }
+        return scheduledTime;
+    }
+    
     protected void addScheduledActivityAtTime(List<ScheduledActivity> scheduledActivities, SchedulePlan plan,
             ScheduleContext context, DateTime scheduledTime) {
```

Comment:
Likewise add...AtTime() should take a LocalDateTime, since we ignore its timezone anyway.